git-fixup

```diff
@@ -45,13 +46,17 @@ function fixup_candidates_lines () {
 # Produce suggestion of commits by taking the latest commit to each file with
 # staged changes
 function fixup_candidates_files () {
-    git diff --cached --name-only | (
-        while read file; do
-            git rev-list $rev_range -- $file \
-                | grep -v -f <(git rev-list -E --grep='^(fixup|squash)' $rev_range -- $file) \
-                | head -n1
-        done
-    ) | sed 's/^/F /g'
+    if test "$show_all" == "false"; then
```

Comment:
It's weird to see this check inside one of the candidate functions. I would think --all would by pass all heuristics and give you the full range but as it stands you still get `fixup_candidate_lines`.

Unless I'm missing something here I would do this check in `print_candidates` instead, in fact, I think this could really just be an alias for a single `git log` call.

Reply:
done